### ISSUE TITLE
Rule S3264: FP when using +=

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#4276]
### Description

[Rule S3264](https://rules.sonarsource.com/csharp/RSPEC-3264): a false positive is raised when chaining events. 

### Repro steps

```
namespace TestCases
{
    using System;

    public class EventChainingDirect
    {
        public event EventHandler Logging;

        public void RunTest()
        {
            EventTestBase eventTestBase = new EventTestBase();
            eventTestBase.Logging += Logging;
            eventTestBase.RunSomethingThatSendAnEvent();
        }
    }

    public class EventChainingLambda
    {
        public event EventHandler Logging;

        public void RunTest()
        {
            EventTestBase eventTestBase = new EventTestBase();
            eventTestBase.Logging += (s, e) => Logging(s, e);
            eventTestBase.RunSomethingThatSendAnEvent();
        }
    }

    public class EventTestBase
    {
        public event EventHandler Logging;

        public void RunSomethingThatSendAnEvent() => Logging(this, EventArgs.Empty);
    }
}
```

### Expected behavior

The issue should not be raised in this context.

### Actual behavior

`EventChainingDirect.Logging` is considered non-compliant although the event is invoked by `RunSomethingThatSendAnEvent`

### Known workarounds

Using a lambda function like in `EventChainingLambda`

### Related information

* C#/VB.NET Plugins version: 8.21

**Comment by MrDaedra** [comment]
We are facing the same issue but not even with direct chaining. For `ProcessMessageAsync` event we get S3264: Remove the unused event 'ProcessErrorAsync' or invoke it.
```C#
internal class ServiceBusMessageProcessor
{
    public event Func<ProcessBrokerMessageEventArgs, Task> ProcessMessageAsync;

    public ServiceBusMessageProcessor(ServiceBusClient serviceBusClient, string topic, string subscriptionName)
    {
        var processor = serviceBusClient.CreateProcessor(topic, subscriptionName, new ServiceBusProcessorOptions());
        processor.ProcessMessageAsync += OnProcessMessageAsync;
    }

    private async Task OnProcessMessageAsync(ProcessMessageEventArgs arg)
    {
        await ProcessMessageAsync?.Invoke(new ProcessServiceBusMessageEventArgs(arg));
    }
}
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1914](https://sonarsource.atlassian.net/browse/NET-1914)

[NET-1914]: https://sonarsource.atlassian.net/browse/NET-1914?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by alex-meseldzija-sonarsource** [comment]
We just released version [10.18](https://github.com/SonarSource/sonar-dotnet/releases/tag/10.18.0.131500) of our analyzers which fixes this issue.
